### ISSUE TITLE
AWS clients should check `isBase64Encoded`

### ISSUE THREAD

**Issue by jmackie** [quirrel-dev/quirrel#896]
## Bug Report

**Current Behavior**

Shipping a quirrel client to Vercel (and presumably any other provider that uses AWS infrastructure) will fail if it receives a base64 encoded body.

**Expected behavior/code**

if `event.isBase64Encoded` then `event.body` should be decoded before passing to `quirrel.respondTo`.

**Environment**

- Quirrel version: v1.7.4
- Node/npm version: Node 16

**Possible Solution**

**Additional context/Screenshots**

Obscure error in Vercel triggered by this:

<img width="1025" alt="image" src="https://user-images.githubusercontent.com/12185627/145452728-9b378044-9d5b-41de-b525-89876500bbb4.png">

**Comment by Skn0tt** [comment]
Hi! That's an interesting error. I was under the impression that Vercel automatically did the base64 conversion for you 🤔
`event.isBase64Encoded` isn't accessible from Vercel's user-facing API, either.

The stack trace says "base64" since `secure-e2ee` uses base64 encoding internally, but I don't think that's what's happening here. Since incoming string passes Quirrel's [signature authentication](https://github.com/quirrel-dev/quirrel/blob/89c08bbfb2cc81994b41fbf0fa8d889201b7a8b8/src/client/index.ts#L542), it seems to me something else is happening here. Could it be that you used the Production Quirrel instance locally during testing, where you didn't have a encryption secret specified, and now the same job is delivered to production where there is an encryption secret?

**Comment by jmackie** [comment]
> I was under the impression that Vercel automatically did the base64 conversion for you

I don't think this assumption is correct. At least not based on my experience (I did some hacky logging of requests to diagnose this issue, and `isBase64Encoded` was there).

Here's how I fixed the issue:

```typescript
import { Queue } from "quirrel/redwood";

const queue = Queue<unknown>("queue", myJobHandlerFunction);

export const handler = (req: any, res: any) => {
  try {
    const { isBase64Encoded = false, body } = req;
    const bodyString = isBase64Encoded
      ? Buffer.from(body, "base64").toString("utf-8")
      : body;

    return queue({ ...req, body: bodyString }, res);
  } catch {
    return queue(req, res);
  }
};
```

**Comment by Skn0tt** [comment]
Ahh, I see! You're running Redwood. I somehow assumed you were running Next or Vercel Serverless Functions, where `isBase64Encoded` is indeed abstracted away from you. But with Redwood, Vercel apparently exposes the raw Lambda API.
